src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifFileWithContentsTests.cs

```diff
@@ -97,35 +183,85 @@ public void SarifFileWithContents_SavesContents()
         {
             var fileDetails = CodeAnalysisResultManager.Instance.FileDetails;
 
-            fileDetails.Should().ContainKey(Key);
+            fileDetails.Should().ContainKey(Key1);
         }
 
         [Fact]
-        public void SarifFileWithContents_DecodesContents()
+        public void SarifFileWithContents_DecodesBinaryContents()
         {
-            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[Key];
+            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[Key2];
             var contents = fileDetail.GetContents();
 
-            contents.Should().Be(Contents);
+            fileDetail.Sha256Hash.Should().Be(ExpectedHashValue2);
+            contents.Should().Be(ExpectedContents2);
         }
 
         [Fact]
-        public void SarifFileWithContents_OpensEmbeddedFile()
+        public void SarifFileWithContents_OpensEmbeddedBinaryFile()
         {
-            var rebaselinedFile = CodeAnalysisResultManager.Instance.CreateFileFromContents(Key);
-
+            var rebaselinedFile = CodeAnalysisResultManager.Instance.CreateFileFromContents(Key2);
+            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[Key2];
             var fileText = File.ReadAllText(rebaselinedFile);
 
-            fileText.Should().Be(Contents);
+            fileDetail.Sha256Hash.Should().Be(ExpectedHashValue2);
+            fileText.Should().Be(ExpectedContents2);
+        }
+
+        [Fact]
+        public void SarifFileWithContents_DecodesTextContents()
+        {
+            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[Key3];
+            var contents = fileDetail.GetContents();
+
+            contents.Should().Be(ExpectedContents1);
+        }
+
+        [Fact]
+        public void SarifFileWithContents_DecodesBinaryContentsWithText()
+        {
+            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[Key4];
+            var contents = fileDetail.GetContents();
+
+            fileDetail.Sha256Hash.Should().Be(ExpectedHashValue2);
+            contents.Should().Be(ExpectedContents2);
+        }
+
+        [Fact]
+        public void SarifFileWithContents_HandlesEmptyBinaryContents()
+        {
+            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[Key5];
+            var contents = fileDetail.GetContents();
+
+            contents.Should().Be(string.Empty);
```

Comment:
Add the hash value to show that you don't skip hashing an empty file -- here, and in the next test.